### PR TITLE
Avoid having multiple measurements of same type at single encounter

### PR DESCRIPTION
**Pull request by anvmn**
https://github.com/Gizra/ihangane/issues/3051

### CODE REVIEW

**Comment by anvmn** [comment]
Selection_1985

File: server/hedley/modules/custom/hedley_admin/scripts/delete-duplicate-measurements.php [hunk 1]
```diff
@@ -20,121 +22,78 @@
...
+        $query->addField('fp', 'field_person_target_id');
+        $query->condition('fp.field_person_target_id', $person_id);
       }
+      $query->condition('et.bundle', $bundle);
+      $query->condition("et.field_{$encounter_type}_target_id", $encounter);
+      $query->orderBy('n.vid', 'DESC');
+      $result = $query->execute()->fetchAllAssoc('entity_id');
+
+      $duplicates = array_keys($result);
+      $first = array_shift($duplicates);
```
Comment (mariano-dagostino) [inline server/hedley/modules/custom/hedley_admin/scripts/delete-duplicate-measurements.php]: @anvmn I don't fully understand the logic behind this array_shift

**Review by mariano-dagostino — APPROVED** [review]
@anvmn Looks good. But could you clarify my question first?

File: server/hedley/modules/custom/hedley_admin/scripts/delete-duplicate-measurements.php (hunk 1 above)
Comment (anvmn) [reply server/hedley/modules/custom/hedley_admin/scripts/delete-duplicate-measurements.php]: @mariano-dagostino 
The idea is that there are several nodes that are duplicates of each other. We want to delete all but one.
`$first = array_shift($duplicates);` pulls first node from that array, and `$duplicates` now got all remaining nodes, which are going to be deleted.

Comment (anvmn) [reply server/hedley/modules/custom/hedley_admin/scripts/delete-duplicate-measurements.php]: @mariano-dagostino ^^

**Comment by anvmn** [comment]
@mariano-dagostino 

Added few code improvements. Please have another look.

**Pull request merged**